pkgs/build-support/fake-uname/default.nix

```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, bash }:
+
+let unameTuple = [
+    /* kernel-name */       "Linux"
+    /* node-name */         "localhost"
+    /* kernel-release */    "4.4.0-not-a-real-version"
+    /* kernel-version */    "#1 SMP PREEMPT Thu Jan  1 00:00:01 GMT 1970"
+    /* machine */           (builtins.head (lib.splitString "-" stdenv.system))
```

Comment:
For the record, `hostPlatform.system` is probably a better way to get what you want. Makes which of the 3 platforms you want explicit, and during cross compilation the `stdenv.*` stuff incorrectly reflects the build platform.